### PR TITLE
app status monitor change for PNA changed case

### PR DESCRIPTION
**Pull request by bisztray**

### CODE REVIEW

**Comment by bisztray** [comment]
Fixes a problem where upon PNA change the app status update is not called

File: application-lib/pkg/monitoring/monitoring.go [hunk 1]
```diff
@@ -77,8 +80,8 @@ func (m *Monitor) Run() {
 			UpdateFunc: func(oldObj, newObj interface{}) {
 				log.Info("Pod changed")
 
-				status := m.GetApplicationStatus()
-				if m.Instance.GetStatus().GetAppStatus() != status {
+				status := m.GetApplicationStatusUsingPodStatus()
+				if m.InstanceAppStatus != status {
```
Comment (howardhalili) [inline application-lib/pkg/monitoring/monitoring.go:84]: i think you also need to update m.InstanceAppStatus after successful update of app specific CR's status.appStatus.

**Review by howardhalili — CHANGES_REQUESTED** [review]

File: application-lib/pkg/monitoring/monitoring.go (hunk 1 above)
Comment (bisztray) [reply application-lib/pkg/monitoring/monitoring.go:84]: Added to line 131 https://github.com/nokia/industrial-application-framework/pull/78/commits/ca6158104ac8c5104fd1107993d8d32d87b319c2

Comment (howardhalili) [reply application-lib/pkg/monitoring/monitoring.go:84]: i don't think it's a good place. i recommend to move it in the else clause (to be added) for the if statement in [line 113](https://github.com/bisztray/industrial-application-framework/blob/ca6158104ac8c5104fd1107993d8d32d87b319c2/application-lib/pkg/monitoring/monitoring.go#L113) of current code, that way, we are sure that the app specific CR status update is successful before updating Monitor.InstanceAppStatus.

**Review by howardhalili — APPROVED** [review]

**Pull request merged**